src/server/frontend_wayland/wl_seat.cpp

```diff
@@ -117,7 +117,7 @@ void mf::WlSeat::ConfigObserver::device_changed(std::shared_ptr<input::Device> c
 {
     if (auto keyboard_config = device->keyboard_configuration())
     {
-        if (*current_keymap != keyboard_config.value().device_keymap())
+        if (!current_keymap->matches(keyboard_config.value().device_keymap()))
         {
             pending_keymap = keyboard_config.value().device_keymap_shared();
         }
```

Comment:
I would be inclined to drop the conditionals here and put the test in `changes_complete()`.

One reason being, that if there are two calls to `device_added()/device_changed()` and the second matches `current_keymap` it gets dropped on the floor.

Reply:
Good idea